### ISSUE TITLE
Multiple API hits on single resource

### ISSUE THREAD

**Issue by cezarykluczynski** [jcabi/jcabi-github#1155]
I would expect that calling something like:

```java
User privateUser = github.users().get(name);
Smart user = new Smart(privateUser);
user.login();
user.avatarUrl().toString();
user.type();
```

will call the user resource once, and then extract stored data when login(), avatarUrl(), and type() are called. Seems like this is not the case, and new API request is performed for every method call. Am I missing something here? Is this the intended behaviour?

It seems like the only option to get all the user data in a single request, and single request is the only choice for me, is to get the raw JSON, and then extract everything from it, bypassing ale the nice getters User.Smart have.

**Comment by dmarkov** [comment]
@yegor256 please do something about this issue

**Comment by cvrebert** [comment]
+100

**Comment by yegor256** [comment]
@cezarykluczynski yes, that's how it is intended to be. The only way to "fix" this is to introduce a new "caching" decorator of `Github`, which you would use like this:

```java
User privateUser = new CachedGithub(github).users().get(name);
Smart user = new Smart(privateUser);
user.login();
user.avatarUrl().toString();
user.type();
```

Pay attention to `new CachedGithub(github)`. See how a similar thing is implemented in, say, https://github.com/jcabi/jcabi-s3/tree/master/src/main/java/com/jcabi/s3/cached
